src/main/kotlin/com/wolt/wm/training/bank/api/CustomerController.kt

```diff
@@ -38,11 +40,39 @@ class CustomerController(private val customerService: CustomerService) {
 
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
-    fun createCustomer(@RequestBody customer: Customer) = customerService.createCustomer(customer)
+    fun createCustomer(@RequestBody customerRequest: CreateCustomerRequest) {
+        val newCustomerId = UUID.randomUUID()
```

Comment:
I wonder if it is better to convert the request object into domain object in the controller or better to do it in service?

Reply:
Controller would be the right place. We can extract this into a Mapper class and call it from the controller.
Services should be callable with internal domain objects, so that we don't need API types when a service call another one.